lib/AppInfo/Application.php

```diff
@@ -115,12 +122,29 @@ public function boot(IBootContext $context): void
                     $session->close();
                     header('Clear-Site-Data: "cache", "storage"');
 
+                    if ($refreshTokensEnabled) {
+                        $logoutUrl = $this->tokenService->getLogoutUrl();
+                    }
+
                     header('Location: '.$logoutUrl);
 
                     exit;
                 });
             }
 
+            if ($refreshTokensEnabled && !$this->tokenService->refreshTokens()) {
+                // See if session is active if autologin is enabled; else logout
```

Comment:
Does this make sense?  Say user clicks "remember me" at SSO and their session there is active for long..

This way, if tokens are expired, autologin will happen against the active session. In case auto redirect is off, we just logout after tokens are expired..